File: Be/property-service/src/main/java/com/property/entity/Apartment.java
package com.property.entity;

import java.util.Set;

import jakarta.persistence.*;

import lombok.*;
import lombok.experimental.FieldDefaults;
import lombok.extern.slf4j.Slf4j;

@Entity
@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Slf4j
@Table(name = "apartment", indexes = {
        @Index(name = "idx_user_id_apartment", columnList = "userId"),
        @Index(name = "idx_city_apartment", columnList = "city"),
        @Index(name = "idx_is_available_apartment", columnList = "isAvailable")
})
public class Apartment extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String apartmentId;

    String userId;

    String name;

    String city;
    String address;
    @Column(columnDefinition = "TEXT")
    String rule;
    @Column(columnDefinition = "TEXT")
    String utility;
    @Column(columnDefinition = "TEXT")
    String description;
    Boolean isAvailable;

    @OneToMany
    @JoinColumn(name = "apartment_id")
    Set<AdditionalCost> additionalCosts;

    @ManyToOne
    @JoinColumn(name = "apartment_type_name", referencedColumnName = "name")
    ApartmentType apartmentType;
}

File: Fe/client/src/dto/request/apartment.ts
import { z } from "zod";

export const apartmentRequest = z
	.object({
		city: z
			.string()
			.min(5, { message: "Vui lòng chọn thành phố" }),
		address: z
			.string()
			.min(10, { message: "Địa chỉ có ít nhất 10 kí tự" })
			.max(100, { message: "Địa chỉ có nhiều nhất 100 6 kí tự" }),
		rule: z.string().max(500, { message: "Quy định có nhiều nhất 500 kí tự" }),
		description: z.string().max(500, { message: "Mô tả có nhiều nhất 500 kí tự" }),
		name: z.string().min(5, { message: "Tên có ít nhất 5 ký tự" }).max(100, { message: "Tên có nhiều nhất 100 kí tự" }),
		utility: z.string().max(500, { message: "Tiện ích có nhiều nhất 500 kí tự" }),
		apartmentType: z.string().min(1, { message: "Hãy chọn loại phòng trọ" }),
	})
	.strict();